tests/test_router_bgp.rb

```diff
@@ -229,6 +229,27 @@ def test_get_asnum
     bgp.destroy
   end
 
+  def test_nsr
+    nsr(setup_default)
+    nsr(setup_vrf)
+  end
+
+  def nsr(bgp)
+    if platform == :nexus || !@vrf[/default/]
+      assert_raises(Cisco::UnsupportedError) do
```

Comment:
For unsupported cases, you should also check `bgp.nsr == false` and `bgp.default_nsr == false`.